test/XRP/serializer.test.ts

```diff
@@ -1405,4 +1410,108 @@ describe('serializer', function (): void {
+    offerSequence.setValue(offerSequenceNumber)
+
+    // WHEN it is serialized.
+    const serialized = Serializer.offerSequenceToJSON(offerSequence)
+
+    // THEN the result is the expiration time.
+    assert.equal(serialized, offerSequenceNumber)
+  })
+
+  it('Serializes an Owner', function (): void {
+    // GIVEN an Owner wrapping an address.
+    const owner = new Owner()
+    owner.setValue(testAccountAddress)
+
+    // WHEN it is serialized.
+    const serialized = Serializer.accountToJSON(owner)
+
+    // THEN the result is the address
+    assert.equal(serialized, testAccountAddress.getAddress())
+  })
+
+  it('Fails to serialize an Owner with no AccountAddress', function (): void {
+    // GIVEN an empty Owner.
+    const owner = new Owner()
+
+    // WHEN it is serialized.
+    const serialized = Serializer.ownerToJSON(owner)
+
+    // THEN the result is undefined.
+    assert.isUndefined(serialized)
+  })
+
+  it('Serializes an EscrowCancel with all fields set', function (): void {
+    // GIVEN an EscrowCancel with all fields set.
+    const offerSequence = new OfferSequence()
+    offerSequence.setValue(offerSequenceNumber)
+
+    const owner = new Owner()
+    owner.setValue(testAccountAddress)
+
+    const escrowCancel = new EscrowCancel()
+    escrowCancel.setOfferSequence(offerSequence)
+    escrowCancel.setOwner(owner)
+
+    // WHEN it is serialized.
+    const serialized = Serializer.escrowCancelToJSON(escrowCancel)
+
+    const expectedJSON: EscrowCancelJSON = {
+      OfferSequence: offerSequenceNumber,
+      Owner: testAccountAddress.toString(),
+      TransactionType: 'EscrowCancel',
+    }
+
+    // THEN the result is as expected.
+    assert.deepEqual(serialized, expectedJSON)
+  })
+
+  it('Fails to serialize an EscrowCancel missing an offerSequence', function (): void {
+    // GIVEN an EscrowCancel that's missing an offerSequence.
+    const owner = new Owner()
+    owner.setValue(testAccountAddress)
+
+    const escrowCancel = new EscrowCancel()
+    escrowCancel.setOwner(owner)
+
+    // WHEN it is serialized.
+    const serialized = Serializer.escrowCancelToJSON(escrowCancel)
+
+    // THEN the result is undefined.
+    assert.isUndefined(serialized)
+  })
+
+  it('Fails to serialize an EscrowCancel missing an owner', function (): void {
+    // GIVEN an EscrowCancel that's missing an owner.
+    const offerSequence = new OfferSequence()
+    offerSequence.setValue(offerSequenceNumber)
+
+    const escrowCancel = new EscrowCancel()
+    escrowCancel.setOfferSequence(offerSequence)
+
+    // WHEN it is serialized.
+    const serialized = Serializer.escrowCancelToJSON(escrowCancel)
+
+    // THEN the result is undefined.
+    assert.isUndefined(serialized)
+  })
+
+  it('Fails to serialize an EscrowCancel with a malformed owner', function (): void {
+    // GIVEN an EscrowCancel with a malformed owner.
+    const owner = new Owner()
+
+    const escrowCancel = new EscrowCancel()
```

Comment:
I think because you haven't set an offer sequence this isn't actually checking that owner is malformed. Instead it is failing out because offerSequence is missing